components/org.wso2.carbon.identity.application.authenticator.fido2/src/main/java/org/wso2/carbon/identity/application/authenticator/fido2/core/WebAuthnService.java

```diff
@@ -867,10 +869,12 @@ private StartRegistrationOptions buildStartRegistrationOptions(User user, boolea
 
     private AuthenticatorSelectionCriteria buildAuthenticatorSelection(boolean requireResidentKey) {
 
-        return AuthenticatorSelectionCriteria.builder()
-                .requireResidentKey(requireResidentKey)
-                .userVerification(PREFERRED)
-                .build();
+        if (requireResidentKey) {
+            return AuthenticatorSelectionCriteria.builder().residentKey(ResidentKeyRequirement.REQUIRED)
+                    .userVerification(PREFERRED).build();
+        }
+        return AuthenticatorSelectionCriteria.builder().residentKey(ResidentKeyRequirement.DISCOURAGED)
```

Comment:
If `requireResidentKey` is false, does it set `ResidentKeyRequirement = DISCOURAGED` previously?
If not, IMO `PREFERRED` would be much suitable. This can have [3 values](https://developers.yubico.com/java-webauthn-server/JavaDoc/webauthn-server-core/2.0.0/com/yubico/webauthn/data/ResidentKeyRequirement.html)

Reply:
Migration docs mention as [Replace requireResidentKey(false) with residentKey(ResidentKeyRequirement.DISCOURAGED)](https://developers.yubico.com/java-webauthn-server/Migrating_from_v1.html#:~:text=Replace%20requireResidentKey(false)%20with%20residentKey(ResidentKeyRequirement.DISCOURAGED).%20Example%3A)